contracts/impl/EnigmaCommon.sol

```diff
@@ -133,6 +133,6 @@ library EnigmaCommon {
     pure
     returns (bytes memory)
     {
-        return (_message.concat(hex"01")).concat(uint64(_arraylength.mul(_typeLength)).toBytesFromUint64());
+        return (_message.concat(hex"01")).concat(uint64(_arraylength.mul(_typeLength.add(9))).toBytesFromUint64());
```

Comment:
I think that the hardcoded solution is never good. It should be a size of byte and size of uint64.

Reply:
@moriaab: while in principle I agree with your statement that hardcoded is no good, I believe this is a place where it is ok for the following reasons:
- By definition, the size of 1 byte is 1, and uint64 is 8 bytes. So as long as the types are hardcoded in that same line, hardcoding the sizes of those types is ok. As far as I understand these are universal truths in the computing world, and I cannot think of a scenario where 1 bytes takes more than 1 byte, or an 8-byte integer takes anything other than 8 bytes.
- we are pushing the limit of the `stack` of smart contracts in many places where we are running out of space in the stack. Defining some of these values as variables/constants to use them later, puts more stress on the stack potentially causing future problems.
- Also, by reducing the number of operations in a smart contract in Solidity, we are thereby reducing the cost of executing the code. While this is not a reason that would stand by itself, this is more of an added benefit after taking the two above into consideration.

So I am ok, leaving it as is.

Reply:
It makes sense, thanks for the explanations.